### DISCUSSION TITLE
How to elegantly convert IDL struct into generated DDS class?

### DISCUSSION THREAD

**Discussion by Octavian-Zhang** [eProsima/Fast-DDS-Gen#225] [Q&A]
I have an existing struct used everywhere in production code:
```
struct HelloWorld
{
    unsigned long index;
    string message;
};
```
Now, I've decided to refactor its cross-machine communication using DDS. I put the exact same struct definition in an IDL file but got a generated DDS datatype class colliding with the existing production code struct definition. 

- Question 1: is it possible to automatically add a name extension to the generated DDS datatype class? 

Moreover, the generated DDS HelloWorld class isn't implicitly convertible with the current HelloWorld struct. 

- Question 2: is it possible to enable the implicit conversion between the DDS datatype class and the original struct without manual field-by-field assignment?

**Comment by JLBuenoLopez** [comment]
@Octavian-Zhang 

1. I am afraid that there is no option to add a name extension to a specific data_type, but if you change the name of your IDL structure, the datatype class would be generated with the new name. Maybe this is enough for your use case.
2. Fast DDS-Gen provides an API to handle datatypes when using Fast DDS. The API specifies the interface classes TypeSupport and TopicDataType for the datatypes defined in the IDL provided. You can always modify the generated code or directly code the specialization yourself without using the auto-generation tool.
